nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -151,8 +159,8 @@ let
       # containerInit and stage2 have to handle this as well.
       exec ${config.systemd.package}/bin/systemd-nspawn \
         --keep-unit \
-        -M "$INSTANCE" -D "$root" $extraFlags \
-        $EXTRA_NSPAWN_FLAGS \
+        -M "$INSTANCE" -D "$root" "$extraFlags" \
+        "$EXTRA_NSPAWN_FLAGS" \
```

Comment:
Since extra flags is a string here, this will work. Normally, especially in stdenv, extraFlags is an array, which would mean we would mess with quoting here.

Reply:
So there's nothing to be done here? Would you do this differently?

For an array, I guess you'd want to do `"${extraFlags[@]}"` instead, but then we'd need to test first whether the variable is an array or a string and act accordingly. I'm not sure that that's needed here currently?

Reply:
yeah, unfortunate naming